sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/UnitTestResultsImportSensorTest.java

```diff
@@ -53,75 +54,32 @@
-    when(results.tests()).thenReturn(42);
-    when(results.skipped()).thenReturn(1);
-    when(results.failures()).thenReturn(2);
-    when(results.errors()).thenReturn(3);
-    when(results.executionTime()).thenReturn(321L);
-
-    UnitTestResultsAggregator unitTestResultsAggregator = mock(UnitTestResultsAggregator.class);
-    SensorContextTester context = SensorContextTester.create(temp.newFolder());
-    AnalysisWarnings analysisWarnings = mock(AnalysisWarnings.class);
-
-    when(unitTestResultsAggregator.aggregate(Mockito.any(WildcardPatternFileProvider.class), Mockito.any(UnitTestResults.class)))
-      .thenReturn(results);
-
-    new UnitTestResultsImportSensor(unitTestResultsAggregator, ProjectDefinition.create(), "cs", "C#", analysisWarnings)
-      .analyze(context, results);
-
-    verify(unitTestResultsAggregator).aggregate(Mockito.any(WildcardPatternFileProvider.class), Mockito.eq(results));
-
-    assertThat(context.measures("projectKey"))
-      .extracting("metric.key", "value")
-      .containsOnly(
-        tuple(CoreMetrics.TESTS_KEY, 42),
-        tuple(CoreMetrics.SKIPPED_TESTS_KEY, 1),
-        tuple(CoreMetrics.TEST_FAILURES_KEY, 2),
-        tuple(CoreMetrics.TEST_ERRORS_KEY, 3),
-        tuple(CoreMetrics.TEST_EXECUTION_TIME_KEY, 321L));
   }
 
   @Test
   public void should_not_save_metrics_with_empty_results() throws Exception {
-    SensorContextTester context = SensorContextTester.create(temp.newFolder());
-
-    UnitTestResultsAggregator unitTestResultsAggregator = mock(UnitTestResultsAggregator.class);
+    UnitTestResultsAggregator aggregator = mock(UnitTestResultsAggregator.class);
+    DotNetPluginMetadata metadata = mockCSharpMetadata();
     AnalysisWarnings analysisWarnings = mock(AnalysisWarnings.class);
-    UnitTestResults results = mock(UnitTestResults.class);
-    when(results.tests()).thenReturn(0);
-    when(results.skipped()).thenReturn(1);
-    when(results.failures()).thenReturn(2);
-    when(results.errors()).thenReturn(3);
-    when(results.executionTime()).thenReturn(null);
-    when(unitTestResultsAggregator.aggregate(Mockito.any(WildcardPatternFileProvider.class), Mockito.any(UnitTestResults.class))).thenReturn(results);
-
-    new UnitTestResultsImportSensor(unitTestResultsAggregator, ProjectDefinition.create(), "cs", "C#", analysisWarnings)
-      .analyze(context, results);
+    UnitTestResults results = new UnitTestResults();
+    results.add(42, 1, 2, 3, null);
+    when(aggregator.hasUnitTestResultsProperty()).thenReturn(true);
+    when(aggregator.aggregate(any())).thenReturn(results);
+    SensorContextTester context = SensorContextTester.create(temp.newFolder());
+    when(aggregator.aggregate(any())).thenReturn(results);
 
-    verify(unitTestResultsAggregator).aggregate(Mockito.any(WildcardPatternFileProvider.class), Mockito.eq(results));
+    new UnitTestResultsImportSensor(aggregator, metadata, analysisWarnings).execute(context);
 
     assertThat(context.measures("projectKey"))
```

Comment:
It would be great to add an explicit assertion that we expect the value for execution time to be missing (since it's passed as `null` in `results.add(42, 1, 2, 3, null);`). 

I think this will make the test (`should_not_save_metrics_with_empty_results`) easier to understand.